src/celengine/astroobj.cpp

```diff
@@ -5,18 +5,61 @@
 #include "astroobj.h"
 #include "category.h"
 
+AstroObject::AstroObject(AstroObject &o)
+{
+    setIndex(o.getIndex());
```

Comment:
you don't need wrapper functions here
just `m_index = o.m_index`

Reply:
Oh yes, we need. See at implementation of `setIndex`.